src/services/getExpectedApy.ts

```diff
@@ -11,16 +11,48 @@ export const getExpectedApy = (
   vt: number, // variable token balance
   margin: number, // margin until the current timestamp (margin + accrued cashflow)
   predictedApr: number, // the predicted variable APY between current and end
+  rateOracleID: number, // the rate oracle ID that suggests what type of APY is used (linear vs. compounding)
 ): [number, number] => {
-  // (rate[end]/rate[current])^(YEAR / (end-current)) - 1 = APY(current, end)
-  // rate[end]/rate[current] = (APY(current, end) + 1)^((end - current)/YEAR)
-  // rate[end]/rate[current] - 1 = (APY(current, end) + 1)^((end - current)/YEAR) - 1
-  const vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1;
+  if (end <= current) {
+    return [0, 0];
+  }
+
+  let vf = 0;
+
+  switch (rateOracleID) {
+    case 1:
+    case 2:
+    case 3:
+    case 4:
+    case 5:
+    case 6:
+    case 7: {
+      // (rate[end]/rate[current])^(YEAR / (end-current)) - 1 = APY(current, end)
+      // rate[end]/rate[current] = (APY(current, end) + 1)^((end - current)/YEAR)
+      // rate[end]/rate[current] - 1 = (APY(current, end) + 1)^((end - current)/YEAR) - 1
+      vf = (1 + predictedApr) ** getAnnualizedTime(current, end) - 1;
```

Comment:
i wonder if even in this case it's worth considering still doing predictedApr * getAnnualizedTime(current, end), i think that's what all the users think when they input in the box, e.g. such that if you input expected apy equail to fixed you get 0 cashflow intuitively as a user, wdyt @0xZenus ?

Reply:
that was one of my thoughts as well but I was thinking to separate this decision from the current situation (i.e. introducing linear APY to the other rate oracles). Happy to do this if you can confirm.

Reply:
hmm, yeah, let's separate these decisions but note this one for a potential future PR?